AppKit/CPWindow/CPWindow.j

```diff
@@ -3844,6 +3814,263 @@ var interpolate = function(fromValue, toValue, progress)
+        [self _removeTrackingArea:trackingAreas[i]];
+}
+
+- (void)_addTrackingArea:(CPTrackingArea)trackingArea
+{
+    var trackingAreaView = [trackingArea view];
+    
+    if (![_trackingAreaViews containsObject:trackingAreaView])
+        [_trackingAreaViews addObject:trackingAreaView];
+    
+    // If CPTrackingAssumeInside option is set, put the tracking area in the _mouseEnteredStack
+    
+    if ([trackingArea options] & CPTrackingAssumeInside)
+        [_mouseEnteredStack addObject:trackingArea];
+}
+
+- (void)_removeTrackingArea:(CPTrackingArea)trackingArea
+{
+    // If mouse is in the tracking area, we remove it from the stack to avoid to fire a future mouseExited event
+    
+    [_mouseEnteredStack removeObjectIdenticalTo:trackingArea];
+    
+    var trackingAreaView = [trackingArea view];
+    
+    [_trackingAreaViews removeObjectIdenticalTo:trackingAreaView];
+}
+
+- (void)_trackEvent:(CPEvent)anEvent
+{
+    var mouseEnteredStack = [],
+        cursorUpdateStack = [],
+        isKeyWindow       = [self isKeyWindow],
+        point             = [anEvent locationInWindow],
+        dragging          = ([anEvent type] !== CPMouseMoved);
+    
+    for (var i = 0; i < _trackingAreaViews.length; i++)
+    {
+        var aView         = _trackingAreaViews[i],
+            trackingAreas = [aView trackingAreas];
+        
+        if ([aView isHidden])
+            continue;
+        
+        for (var j = 0; j < trackingAreas.length; j++)
+        {
+            var aTrackingArea   = trackingAreas[j],
+                trackingOptions = [aTrackingArea options];
+            
+            if (dragging && !(trackingOptions & CPTrackingEnabledDuringMouseDrag))
+                continue;
+            
+            if ((trackingOptions & CPTrackingActiveAlways) ||
+                (trackingOptions & CPTrackingActiveInActiveApp) ||
+                ((trackingOptions & CPTrackingActiveInKeyWindow) && isKeyWindow) ||
+                ((trackingOptions & CPTrackingActiveWhenFirstResponder) && isKeyWindow && (_firstResponder === aView)))
+            {
+                if (!(CGRectContainsPoint([aTrackingArea actualRect], point)))
+                    continue;
+                
+                [mouseEnteredStack addObject:aTrackingArea];
+                
+                if ([_mouseEnteredStack containsObject:aTrackingArea])
+                {
+                    // Mouse was already in this rect so it's a mouseMoved
+                    
+                    if (!dragging && (trackingOptions & CPTrackingMouseMoved))
+                        [[aTrackingArea owner] mouseMoved:anEvent];
+                }
+                else
+                {
+                    // Mouse was not in this rect so it's a mouseEntered
+                    
+                    if (trackingOptions & CPTrackingMouseEnteredAndExited)
+                        [[aTrackingArea owner] mouseEntered:[CPEvent enterExitEventWithType:CPMouseEntered
```

Comment:
OK. I think my understanding is a "cross language semantic interpretation"  ;-)
Because, in French, should can sometimes mean must, mainly when speaking to children... or (and this is the scary one) when your wife talks to you...  ;-)
I'll do the check.